### PR TITLE
Split Button 

### PR DESCRIPTION
**Pull request by isacoder**
### Description

New Split Button with css external features

 ### Considerations on Implementation

All colors and sizes for height and paddings have been added as the Figma design, this feature has been useful for tuning the Figma to code process.

**Colors variables were updated**
Warning -> Caution
Error -> Warning

 ### Reviewing/Testing steps
Running on storybook on Buttons

Theme files for example and storybook should be equal, I will update in both any request.
This files might be the most important to review since is the first iteration of customization at Component level.

Please review the colors in light and dark mode I have a few considerations about it.

**In light mode**
- Primary active border top looks a bit too dark.
- Secondary text is too dimm in my opinion. Active background contrast is not so notorious, so it's might be hard for the user to know it was pressed.

#### Screenshot

Screenshot 2024-06-27 at 7 05 42

[Video of interaction](https://drive.google.com/file/d/1cplY4wf6hw1AdzkTqy8XbH5RledDyQb-/view?usp=sharing)

### CODE REVIEW

**Review by atomworks — APPROVED** [review]
Looks great, thanks. Few points of feedback but it's up to you if you'd like to fix that here or get this merged as it stands and follow up on another PR.

- When you click a sub-option, it becomes the main top option. Can we make this optional behaviour? It's useful to have but I think commonly we'll want to keep that top option pinned as the most instances will be in things like table rows where the main action should always be at hand. The dynamic change will be more useful in niche workflows.
- There are a few notes in the code. Mostly thoughts or just trivial things to tidy. I think just one question that might need some (very minor) code changes.
- I want to get theme-variables.css in to the `ui-lib` folder moving forward. We've discussed it a bit before. It should be part of that core library. At the moment we have two copies of the file in places that are for examples and not production use. 

Other thoughts:
- I know there are some styling issues. Let's follow up on that separately rather than delay here.
- The button lives in the Form section but as we use it many other places, is there a better place to put it perhaps? It feels like there is a better place but then I know if we think where do normal Buttons go, it starts to get a bit confusing. So, more of a thought for now.

File: packages/ui-lib/src/Form/molecules/MultiActionButton.tsx [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+`;
+
+const validateMaxWidth = (btnTextMaxWidth: number| null | undefined, textMaxWidth?: string) : string | undefined  => {
+
+  if(textMaxWidth)
+    return textMaxWidth;
+
+  if(btnTextMaxWidth)
+    return `${btnTextMaxWidth - 20}px`; // - Remove toggle con size
```
[the remark this answers is not in the document]
Comment (isacoder) [reply packages/ui-lib/src/Form/molecules/MultiActionButton.tsx:101]: Well it's in the comment but I had a typo 🤦‍♀️   is the size of the toggle icon button. I will make it a constant and name it to make it more Obvious.

File: packages/ui-lib/src/Form/atoms/MultiButtonOption.tsx [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+  noBorderTop = false,
+  textMaxWidth='',
+  size='normal',
+  ...props
+}) => {
+
+  return(
+    <StyledButton {...{noBorderTop}} {...props}>
+      <LeftIconWrapper isAscendingIcon={icon === 'FilterAscending'} >
+        {isLoading ? <Spinner size={buttonSpinnerSize(size)} styling={design} /> : <Icon icon={icon} />}
```
[the remark this answers is not in the document]
Comment (isacoder) [reply packages/ui-lib/src/Form/atoms/MultiButtonOption.tsx:134]: In this case the styling is for the spinner, styling/design word are useful to determinate the style name between  components and sub components. I think we can keep using "primary", "secondary" as base styles that are understandable from the css external file.

File: packages/ui-lib/src/Form/atoms/MultiButtonOption.tsx [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+const OptionText = styled.div`
+    font-family: var(--font-ui);
+    font-size: var(--button-font-size);
+    overflow: hidden;
+    white-space: nowrap;
+    text-overflow: ellipsis;
+    text-align: center;
+    font-style: normal;
+    font-weight: 600;
+    line-height: normal;
```
[the remark this answers is not in the document]
Comment (isacoder) [reply packages/ui-lib/src/Form/atoms/MultiButtonOption.tsx]: I have. updated to  be indented as the rest of the code.

**Comment by isacoder** [comment]
> Looks great, thanks. Few points of feedback but it's up to you if you'd like to fix that here or get this merged as it stands and follow up on another PR.
> 
> * When you click a sub-option, it becomes the main top option. Can we make this optional behaviour? It's useful to have but I think commonly we'll want to keep that top option pinned as the most instances will be in things like table rows where the main action should always be at hand. The dynamic change will be more useful in niche workflows.
> * There are a few notes in the code. Mostly thoughts or just trivial things to tidy. I think just one question that might need some (very minor) code changes.
> * I want to get theme-variables.css in to the `ui-lib` folder moving forward. We've discussed it a bit before. It should be part of that core library. At the moment we have two copies of the file in places that are for examples and not production use.
> 
> Other thoughts:
> 
> * I know there are some styling issues. Let's follow up on that separately rather than delay here.
> * The button lives in the Form section but as we use it many other places, is there a better place to put it perhaps? It feels like there is a better place but then I know if we think where do normal Buttons go, it starts to get a bit confusing. So, more of a thought for now.

Updated the requested files and the interaction. We have change the name of this button from multi action button to split button to reflect those changes.

Like we talked over slack I will release this component with this changes and if something is missing will iterate on a different PR

**Pull request merged**